### PR TITLE
Feature: Adds ID property

### PR DESCRIPTION
**Pull request by ajmeese7**
## Background

I use `svelte-simple-modal` with SvelteKit. In #92 the suggested implementation of my need to fade between content in the modal required a way to reference the Modal elsewhere in my codebase.

## Currently Observed Behavior

There is no easy, clean way to reference a specific modal as it stands currently.

## New Behavior

This PR introduces a way to set the modal ID, allowing for easy reference in any external CSS/JavaScript.

**Demo:** https://svelte.dev/repl/ab14f5620f10465b83150c132e361c57?version=3.55.1

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -228,33 +228,34 @@ The `ariaLabel` is automatically omitted when `ariaLabelledBy` is specified.
...
+| ------------------------- | ---------------------------------------------------- | ------------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| **show**                  | Component \| null                                    | `null`              | A Svelte component to show as the modal. See [Store API](#store-api) for details.                                                                                                                                                                           |
+| **id**                    | string \| null                                       | `null`              | Element ID to be applied to the modal. Useful for interfacing with the modal from JavaScript or CSS.                                                                                                                                                        |
```
Comment (flekschas) [inline README.md:234]:
```suggestion
| **id**                    | string \| null                                       | `null`              | Element ID to be assigned to the modal's root DOM element.                                                                                                                                                        |
```

File: src/Modal.svelte [hunk 2]
```diff
@@ -65,6 +65,12 @@
    */
   export let show = null;
 
+  /**
+   * ID to reference the modal in JavaScript/CSS
```
Comment (flekschas) [inline src/Modal.svelte:69]:
```suggestion
   * Element ID assigned to the modal's root DOM element
```

File: src/Modal.svelte [hunk 3]
```diff
@@ -470,6 +477,7 @@
       style={cssWindowWrap}
     >
       <div
+        id={state.id}
```
Comment (flekschas) [inline src/Modal.svelte:480]: To be more generic, the ID should be assigned to the root DOM element. I.e., the background `<div />`. Alternatively, we could also add ID props for each element like we have for the class properties. However, this seems unnecessary as one can easily query for a specific modal element if one has the ID of the root element and the class names.

E.g.,

```svelte
<script>
	import Content from './Content.svelte';
  import Modal from './Modal.svelte';
	import Popup from './Popup.svelte';
	import { modal } from './stores.js';

  const id = 'my-modal';
  const classWindow = 'my-window';

	modal.subscribe((value) => {
		if (!value) return;
		requestAnimationFrame(() => {
			console.log(document.querySelector(`#${id} .${classWindow}`));
		})
	});
</script>

<Modal show={$modal} id={id} classWindow={classWindow}>
	<Content />
</Modal>
```

File: types/Modal.svelte.d.ts [hunk 4]
```diff
@@ -86,6 +87,12 @@ export interface ModalProps {
    */
   show?: Component | null;
 
+  /**
+   * ID to reference the modal in JavaScript/CSS
+   * @default null
+   */
+  id?: string | null;
+
```
Comment (flekschas) [inline types/Modal.svelte.d.ts:95]: Are those auto-generated via `npm run build`? If so, after the above changes you'd have to run the script again to pick up the adjusted text.

File: README.md [hunk 5]
```diff
@@ -3,27 +3,27 @@
 </h1>
 
 <div align="center">
-  
-  **A simple, small, and content-agnostic modal for [Svelte](https://svelte.dev).**
-
```
Comment (flekschas) [inline README.md:8]: Please undo these and the following unnecessary removal of indentations.

**Review by flekschas — CHANGES_REQUESTED** [review]
Thank you for putting together this PR! I think this is a useful new feature. However, I think the ID should be assigned to the root element to provide more flexibility in accessing all modal-related DOM elements.

File: README.md (hunk 5 above)
Comment (ajmeese7) [reply README.md:8]: My bad, guess my autolinter caught em. Will fix!

File: types/Modal.svelte.d.ts [hunk 6]
```diff
@@ -22,10 +19,11 @@ export type Styles = Record<string, string | number>;
 
 export type TransitionFn = (
   node: Element,
-  parameters: BlurParams | FadeParams | FlyParams | SlideParams
```
Comment (flekschas) [inline types/Modal.svelte.d.ts:25]: Can you give me write access to your to your PR so I can fix the removal of these types?

**Review by flekschas — CHANGES_REQUESTED** [review]
Looks good now. But I have to fix the removal of transition param types before we can merge it.

**Comment by ajmeese7** [comment]
Unfortunately GitHub [doesn't allow](https://github.com/orgs/community/discussions/5634#discussioncomment-2968070) me to have the maintainer checkbox since the PR was made from an organizational fork, I'll go back and revert the changes the autolinter made real quick.

**Comment by flekschas** [comment]
No worries. I'll merge this in and then fix it on master

**Comment by flekschas** [comment]
New version (v1.5.0) is pushed. Should be available in a few minutes.

**Pull request merged**